docs/spec/annotations.rst

```diff
@@ -353,10 +349,14 @@ types cannot be specified::
 Annotating instance and class methods
 -------------------------------------
 
-In most cases the first argument of class and instance methods
-does not need to be annotated, and it is assumed to have the
-type of the containing class for instance methods, and a type object
-type corresponding to the containing class object for class methods.
+In most cases the first argument of instance and class methods
+(conventionally named ``self`` or ``cls``) does not need to be annotated.
+
+If the argument is not annotated, then for instance methods it is
+assumed to have the type of the containing class or :ref:`Self
+<self>`, and for class methods the type object type corresponding to
+the containing class object or ``type[Self]``.
```

Comment:
```suggestion
If the argument is not annotated, then for instance methods it may be
inferred to have either the type of the containing class, or the type :ref:`Self
<self>`. For class methods it may be inferred to have either the type object
type corresponding to the containing class object, or ``type[Self]``.
```